File: cle_factorisable/solve.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int isprime(unsigned long n){
    if(n<2){
        return 0;
    }
    if(n==2){
        return 1;
    }
    for(int i=3; i<sqrt(n)+1; i+=2){
        if(n%i == 0){
            return 0;
        }
    }
    return 1;
}

int factors(unsigned long n){
    for(unsigned long x=1; x<sqrt(n)+1; x++){
        if(n%x==0 && n%(n/x)==0 && isprime(x) && isprime(n/x)){
            printf("%lu %lu",x,n/x);
            return 1;
        }
    }
    return 0;
}

int main(){
    factors(407756900493095869);

}

File: cle_factorisable/solve.py
import tqdm
from math import sqrt
def isprime(n): 
    n = abs(int(n))
    if n < 2:
        return False
    if n == 2: 
        return True    
    if not n & 1: 
        return False
    for x in range(3, int(sqrt(n))+1, 2): 
        if n % x == 0:
            return False
    return True

def factors(l):
    n = int(l)
    for x in tqdm.tqdm(range(1,int(sqrt(n))+1)):
        if n%x == 0 and n%(n//x)==0 and isprime(x) == True and isprime(n//x) == True:
            return [x, n//x]

print(factors("407756900493095869"))